### PR TITLE
Add asarray

### PR DESCRIPTION
**Pull request by rsokl**
The first of many convenience functions for interchanging between arrays and tensors

Also adds `tensors` hypothesis strategy

### CODE REVIEW

File: src/mygrad/numpy_compat/array_creation.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    """Convert the input to an array.
+
+    This docstring is adapted from that of ``numpy.asarray``
+
+    Parameters
+    ----------
+    a : array_like
+        Input data, in any form that can be converted to an array.  This
+        includes lists, lists of tuples, tuples, tuples of tuples, tuples
+        of lists and ndarrays.
```
Comment (rsokl) [inline src/mygrad/numpy_compat/array_creation.py:18]:
```suggestion
        of lists, ndarrays, and mygrad-tensors.
```

File: src/mygrad/numpy_compat/array_creation.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+def asarray(a, dtype=None, order=None) -> np.ndarray:
+    """Convert the input to an array.
+
+    This docstring is adapted from that of ``numpy.asarray``
+
+    Parameters
+    ----------
+    a : array_like
+        Input data, in any form that can be converted to an array.  This
```
Comment (rsokl) [inline src/mygrad/numpy_compat/array_creation.py:16]:
```suggestion
        Input data, in any form - including a mygrad tensor - that can be converted to an array. This
```

File: tests/custom_strategies/test_strategies.py [hunk 3]
```diff
@@ -196,3 +199,75 @@ def test_arbitrary_indices_strategy(a, data):
...
     # rules, numpy will raise an error
     a[index]
+
+
+@pytest.mark.parametrize("constant", [st.booleans(), None])
+def test_tensors_handles_constant_strat(constant):
+    constants = []
+    kwargs = dict(dtype=np.int8, shape=(2, 3))
+    if constant is not None:
+        kwargs["constant"] = constant
```
Comment (petarmhg) [inline tests/custom_strategies/test_strategies.py:209]: What is the purpose of the `None`? If `constant` is not specified in the strategy, it uses `st.booleans` anyway. Wouldn't it make more sense to use `[True, False, None]` in the `parametrize` decorator?

File: setup.cfg [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [flake8]
-exclude = docs/*, src/mygrad/__init__.py
+exclude = docs/*, src/mygrad/__init__.py, src/mygrad/numpy_compat/__init__.py
```
[the remark this answers is not in the document]
Comment (rsokl) [reply setup.cfg:2]: It was mad that I imported `asarray` into that namespace without using it there.

File: tests/custom_strategies/test_strategies.py (hunk 3 above)
Comment (rsokl) [reply tests/custom_strategies/test_strategies.py:209]: This just ensures that `None` has the same behavior as specifying `st.booleans()`. The static constant  cases are tested below.

**Comment by rsokl** [comment]
> Could we put up an issue to keep track of which functions have been added and which have not?

Yeah, this would be useful. I don't have a definitive list in mind. `astensor` will definitely be one, but I am still coming up with ideas for others. I should have made `asarray` and `astensor` long ago.. is would have saved us so many if-clauses :/

File: tests/custom_strategies/__init__.py [hunk 5]
```diff
@@ -70,6 +73,155 @@ def _check_min_max(min_val, min_dim, max_dim, param_name, max_val=None):
...
+
+    constant : Union[bool, st.SearchStrategy[bool]]
+        Specifies ``tensor.constant``. Default is :func:`~hypothesis.strategies.booleans`
+
+    include_grad: bool, optional (default=False)
+        If True, draw an array to be set to `tensor.grad`
+
+    grad_dtype: Optional[Any]
+        The datatype of the drawn gradient. Defaults to the dtype of the host tensor.
+
```
Comment (rsokl) [inline tests/custom_strategies/__init__.py:134]:
```suggestion
    grad_dtype: Optional[Any]
        The datatype of the drawn gradient. Defaults to the dtype of the host tensor.
        Specifying ``grad_dtype``, while ``include_grad`` is False, will raise an error.

```

File: tests/custom_strategies/__init__.py [hunk 6]
```diff
@@ -70,6 +73,156 @@ def _check_min_max(min_val, min_dim, max_dim, param_name, max_val=None):
...
+        If True, draw an array to be set to `tensor.grad`
+
+    grad_dtype: Optional[Any]
+        The datatype of the drawn gradient. Defaults to the dtype of the host tensor.
+        Specifying ``grad_dtype``, while ``include_grad`` is False, will raise an error.
+
+    grad_elements_bounds: Optional[Tuple[int, int]]
+        The min and max bounds used to draw the gradient's elements.
+        Defaults to (-10, 10)
+
```
Comment (rsokl) [inline tests/custom_strategies/__init__.py:139]:
```suggestion
    grad_elements_bounds: Optional[Tuple[int, int]]
        The min and max bounds used to draw the gradient's elements.
        Defaults to (-10, 10)
        Specifying ``grad_element_bounds``, while ``include_grad`` is False, will raise an error.

```

File: tests/custom_strategies/test_strategies.py (hunk 3 above)
Comment (petarmhg) [reply tests/custom_strategies/test_strategies.py:209]: Ahh I see :+1:

**Comment by rsokl** [comment]
@petarmhg added the docs - should be good to go if that is everything

**Pull request merged**